File: molecule/default/tests/test_default.rb
# frozen_string_literal: true

# rubocop:disable LineLength

describe file('/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin') do
  its('type') { should eq :directory }
  it { should be_directory }
end

describe file('/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin/python3.7') do
  it { should exist }
end

describe file('/opt/mongodbtoolchain/v3/bin') do
  its('type') { should eq :directory }
  it { should be_directory }
end

describe file('/opt/mongodbtoolchain/v3/bin/python3.7') do
  it { should exist }
end

describe command('/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin/python3.7 --version') do
  its('stdout') { should eq "Python 3.7.0\n" }
  its('exit_status') { should eq 0 }
end

describe command('/opt/mongodbtoolchain/v3/bin/python3.7 --version') do
  its('stdout') { should eq "Python 3.7.0\n" }
  its('exit_status') { should eq 0 }
end

# rubocop:enable LineLength

File: molecule/default/tests/test_default.py
import os

import testinfra.utils.ansible_runner

testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
    os.environ['MOLECULE_INVENTORY_FILE']
).get_hosts('all')


def test_toolchain_version(host):
    assert host.file("/opt/mongodbtoolchain/toolchain_version").exists
    assert host.file("/opt/mongodbtoolchain/toolchain_version").contains("testing_mongo_sha")


def test_revisions(host):
    assert host.file("/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin").exists
    assert host.file("/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin").is_directory
    assert host.file("/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin/python3.7").exists


def test_v3(host):
    assert host.file("/opt/mongodbtoolchain/v3/bin").exists
    assert host.file("/opt/mongodbtoolchain/v3/bin").is_directory
    assert host.file("/opt/mongodbtoolchain/v3/bin/python3.7").exists


def test_python(host):
    cmd = host.run("/opt/mongodbtoolchain/revisions/test_mongodbtoolchain/v3/bin/python3.7 --version")
    assert cmd.stdout == """Python 3.7.0
"""
    assert cmd.succeeded
    cmd = host.run("/opt/mongodbtoolchain/v3/bin/python3.7 --version")
    assert cmd.stdout == """Python 3.7.0
"""
    assert cmd.succeeded
